File: testsuite/tests/ada/0007-tagged-types/cpp/main.cpp
#include "test.h"
#include "test_priv.h"
#include "test_publ.h"

int main() {
    test::Root root(1, 2);
    test::Rec recr(1, 2);

    root.p1();

    test::Child child(3, 4, 5);
    test::Rec recc(3, 4);

    child.p1();

    root.p2(recr);
    child.p2(recc);

    test::OtherChild other_child(3, 4, 5);
    other_child.p1();

    test::Root &foo = other_child;
    foo.p1();

    test::p_root(root);
    test::p_root(child);
    test::p_root(other_child);

    test::priv::PrivRoot priv;
    priv.foo();

    test::publ::Publ publ(1, 3);
    publ.foo();
}

File: testsuite/tests/ada/0007-tagged-types/java/Main.java
import com.adacore.libtest.test.Child;
import com.adacore.libtest.test.OtherChild;
import com.adacore.libtest.test.priv.PrivRoot;
import com.adacore.libtest.test.publ.Publ;
import com.adacore.libtest.test.Rec;
import com.adacore.libtest.test.Root;
import com.adacore.libtest.test.TestPackage;

import java.util.concurrent.CompletableFuture;
import java.util.concurrent.TimeUnit;

public class Main {

    public static void main(String[] args) {
        Root root = new Root(1, 2);
        Rec recr = new Rec(1, 2);

        root.p1();

        Child child = new Child(3, 4, 5);
        Rec recc = new Rec(3, 4);

        child.p1();

        root.p2(recr);
        child.p2(recc);

        OtherChild otherChild = new OtherChild(3, 4, 5);
        otherChild.p1();

        Root foo = otherChild;
        foo.p1();

        TestPackage.pRoot(root);
        TestPackage.pRoot(child);
        TestPackage.pRoot(otherChild);

        PrivRoot priv = new PrivRoot();
        priv.foo();

        Publ publ = new Publ(1, 3);
        publ.foo();
    }
}
